app/features/transactions/transaction.ts

```diff
@@ -195,6 +196,10 @@ export type Transaction = {
    * Transaction details.
    */
   details: object;
+  /**
+   * Notifications for the transaction.
+   */
+  notifications: Notification[];
```

Comment:
this is wrong and not what I had in mind initially. I was thinking that you are loading notifications in some view and then for each you need related transaction data and that you were initially pulling the data separately and joining so I suggested to make notification contain it's payload (transaction)

I don't like any of the proposed approaches. I need to think a bit and propose alternative